ios/viewmodels/view-model-reuse.md

```diff
@@ -0,0 +1,167 @@
+# View Model Reuse
+
+Sometimes a view might be generic enough that it would need to be repurposed for multiple models. The most compelling use case is for a views that are generic enough that they are used in multiple different places:
+
+![ProfileViewController](profile.png)
+![ReviewViewController](review.png)
+
+Let's call these two views ProfileViewController and ReviewViewController respectively.
+
+The table view cell pictured here is the same in both cases, and we would ideally like to reuse the table view cell and it's xib. In both cases, we are trying to abstract view logic out of the table view cell. However, the models that drive these two cells are very different. Let's explore two approaches of view models that can be used for this view.
+
+### Concrete view model class
+
+This simpler of the two approaches just has a concrete class for the view, which is how we would do it if the cell only had to display one kind of model.
+
+```swift
+
+struct TitleMessageCellViewModel {
+  let title: String
+  let message: String
+  let showsEditButton = false
+}
+
+class TitleMessageCell: UITableViewCell {
+  @IBOutlet private weak var titleLabel: UILabel!
+  @IBOutlet private weak var messageLabel: UILabel!
+
+  private func configure(viewModel: TitleMessageCellViewModel) {
+    titleLabel.text = viewModel.title.uppercaseString
+    messageLabel.text = viewModel.message
+    editButton.hidden = !viewModel.showsEditButton
+  }
+}
+
+```
+
+An implementation for ProfileViewController might look like this:
+
+```swift
+
+// ProfileViewController.swift
+
+func viewModel(forIndex index: Int) {
+  switch index {
+    case 0:
+      return TitleMessageCellViewModel(title: "First Name", message: profile.firstName)
+    case 1:
+      return TitleMessageCellViewModel(title: "Surname", message: profile.lastName)
+    ...
+  }
+}
```

Comment:
I don't know about this point - I have never done view models in that manner(where one view model marshalls another) - can other people commment?